### ISSUE TITLE
Test multiple images with custom weights and save result in csv?

### ISSUE THREAD

**Issue by dsbyprateekg** [AlexeyAB/darknet#5802]
Hi,

In Kaggle or any other other hackethon the submission file is in csv file where one column is test image id or name and second column is prediction. 
So is there any command which will test the image folder and should be able to save the results in a csv file?

I have tried following command for testing my test folder images and save result in a txt file-
![test_multiple](https://user-images.githubusercontent.com/30830541/83410757-9d262980-a434-11ea-9490-fb1f28a61d9f.JPG)

And getting following cannot load image line in the console-
![result_cannot_load](https://user-images.githubusercontent.com/30830541/83410786-ad3e0900-a434-11ea-91b9-4506256a6889.JPG)

**Comment by AlexeyAB** [comment]
Show example of csv-file with several image with several detections.

----

You forgot symbols `<` and `>` in your command https://github.com/AlexeyAB/darknet#how-to-use-on-the-command-line

To process a list of images data/train.txt and save results of detection to result.txt use:
`darknet.exe detector test cfg/coco.data cfg/yolov4.cfg yolov4.weights -dont_show -ext_output < data/train.txt > result.txt`

Pseudo-lableing - to process a list of images data/new_train.txt and save results of detection in Yolo training format for each image as label <image_name>.txt (in this way you can increase the amount of training data) use: `darknet.exe detector test cfg/coco.data cfg/yolov4.cfg yolov4.weights -thresh 0.25 -dont_show -save_labels < data/new_train.txt`

**Comment by dsbyprateekg** [comment]
@AlexeyAB I added symbols < and > with test command `.\darknet.exe detector test data/obj.data cfg/yolo-obj.cfg backup/yolo-obj_final.weights -dont_show -ext_output < data/test.txt > result.txt` but getting error-
![seudo_error](https://user-images.githubusercontent.com/30830541/83415951-964fe480-a43d-11ea-8913-24c555cc41b5.JPG)

**Comment by AlexeyAB** [comment]
crate `run.cmd` file with this content
```
darknet.exe detector test data/obj.data cfg/yolo-obj.cfg backup/yolo-obj_final.weights -dont_show -ext_output < data/test.txt > result.txt 

pause
```
and double click on this file

**Comment by dsbyprateekg** [comment]
Thanks @AlexeyAB .
Now it has created the txt file.
[result.txt](https://github.com/AlexeyAB/darknet/files/4711694/result.txt)

Is there any way I can get only image name and it's predicted target class name?

**Comment by AlexeyAB** [comment]
Don't use `-ext_output` flag

**Comment by dsbyprateekg** [comment]
I removed this flag from the run.cmd but it is generating same results in result.txt.

**Comment by AlexeyAB** [comment]
Show example of required format

**Comment by dsbyprateekg** [comment]
This is screenshot of the sample submission csv file-
![sample_submission](https://user-images.githubusercontent.com/30830541/83476374-32b1cf80-a4ae-11ea-90eb-a7e6d85c9064.JPG)

**Comment by AlexeyAB** [comment]
Do you need detector or classifier?

**Comment by dsbyprateekg** [comment]
I have found a way to make the submission file.
I used following command to save the output in json file-
`./darknet detector test data/obj.data cfg/yolo-obj.cfg backup/yolo-obj.weights -ext_output -dont_show -out data/dance_result.json < data/test.txt`

After that I have written a Python script to fetch required two columns with their values and save the result in csv file.

Once new release of OpenCV will come, I will use cv2 for the same.
Thanks!
